### ISSUE TITLE
String freeze

### ISSUE THREAD

**Issue by thebnich** [mozilla/firefox-ios#683]
As we get closer to shipping, we're planning to freeze all strings in the product, including a11y strings. The string freeze allows us to dispatch all strings in the product to people that can translate across different language by the time we launch v1.

@dusek, what's your opinion on the status of a11y right now? Are there any labels/hints that still need to be updated? It sounds like we want to do the string freeze very soon, so consider any such changes high priority if you're planning on taking them. Sorry for the short notice!

**Comment by thebnich** [comment]
Also, here's the discussion thread for anyone interested: https://mail.mozilla.org/pipermail/mobile-firefox-dev/2015-July/001279.html

**Comment by dusek** [comment]
> @dusek, what's your opinion on the status of a11y right now? Are there any labels/hints that still need to be updated? It sounds like we want to do the string freeze very soon, so consider any such changes high priority if you're planning on taking them. Sorry for the short notice!

@thebnich thanks for the info. I will ASAP address review comments in Pull Requests that you reviewed and contain new strings. I submitted one new PR today (has new strings too). I will give it a thought if there is anything more that should be adjusted w.r.t. a11y that would influence strings.

**Comment by dusek** [comment]
@thebnich do we expect to get v1.1 out before (or together with), or after iOS 9 (say half of September)? If after, I would make one new PR today (for stuff that works only on iOS 9 and needs localized strings).

Also I will try to address the https://github.com/mozilla/firefox-ios/pull/497 today, seems like one recent PR for long-pressing URL bar made some more involved adjustments necessary than just mere textual conflict resolution.

Also, as I have at least one more PR in mind (intro view) that could need new strings but I will definitely not have it ready this week – could I submit a PR that would "preemptively" add only the new strings I am contemplating (in some variables or returned by functions) so that they would make it into the string freeze and be translated, so that I could later (say 1-2 weeks from now) use those variables with localized messages when actually making the another (proper) PR with the full change?

**Comment by thebnich** [comment]
> @thebnich do we expect to get v1.1 out before (or together with), or after iOS 9 (say half of September)?

By v1.1, do you mean v1, or the release following v1? I think we're shooting to release v1 before iOS 9 (hopefully August). I don't think there's an estimated date for the following release yet, but I imagine it will be after iOS 9.

> Also, as I have at least one more PR in mind (intro view) that could need new strings but I will definitely not have it ready this week – could I submit a PR that would "preemptively" add only the new strings I am contemplating (in some variables or returned by functions) so that they would make it into the string freeze and be translated, so that I could later (say 1-2 weeks from now) use those variables with localized messages when actually making the another (proper) PR with the full change?

A preemptive strings PR sounds fine to me, especially if you're estimating a 2 week delay for the actual change. Let's see if @st3fan agrees. Thanks!

**Comment by dusek** [comment]
@thebnich Right now, I see these changes that I would like for v1 and that require new strings:

* intro "first run" wizard (using "preemptive" strings if OK'ed - the real full change requires more design and coding work than usual, definitely not fully doable tomorrow)
* add a11y label to search engine image in search suggestions
* revise a11y hints referring to tapping to refer to double-tapping instead - reload/stop button
* if everything goes well, then also Switch Control-related changes that will speak things like "navigation bar", "home panel" for the respective group of items (starting in iOS 9, does not work on iOS 8)

I will be able to do that tomorrow, July 10 (let's say until 21:00 PM CEST = 15:00 PM EDT). Is that soon enough?

Btw. I am unable to test localized Firefox (wanted to try Czech), I filed [Bug 1182306 - l10n cannot be tried out](https://bugzilla.mozilla.org/show_bug.cgi?id=1182306). I would like to give localized Firefox a try tomorrow if I knew how to make it work (run it).

**Comment by thebnich** [comment]
Your plan sounds good to me, though I'm curious what changes you're anticipating for the "first run" wizard. Planning on giving some accessibility-related hints as part of the walkthrough, perhaps?

I'm not too familiar with our localization scripts -- you might want to try pinging or **NEEDINFO**'ing @stefan.

**Comment by dusek** [comment]
I was thinking two things w.r.t. strings: have some descriptions for the images (like for the first screen "tab bar button is at the top of the screen next to address bar, showing tab count"), and then scroll status string "Introductory slide <n> of 3 ". As I have said, I have not yet made up my mind on how to best make the experience accessible. It is quite important as it is (just like for the sighted users) the very first experience with the app. But right now, I think it is not possible to go from the first page of the wizard to the second. Then I also wanted to make escape gesture perform the "Start browsing" action (i.e. dismissing the wizard).

I now NEEDINFO'ed @st3fan (not @stefan ;-)) in the bug.

**Comment by dusek** [comment]
@thebnich I am very sorry but I hate to say I am running late, won't make it till 15:00 EDT as originally promised, more like 16:00 or 16:30. Sorry for any inconvenience.
